casper/src/main/scala/coop/rchain/casper/ReportingCasper.scala

```diff
@@ -1,15 +1,25 @@
 package coop.rchain.casper
 
 import cats.effect.Sync
+import coop.rchain.casper.ReportingCasper.Report
 import coop.rchain.models.BlockHash.BlockHash
 
 trait ReportingCasper[F[_]] {
-  def trace(hash: BlockHash): F[String]
+  def trace(hash: BlockHash): F[Report]
 }
 
 object ReportingCasper {
+  sealed trait RhoEvent
+  final case class DummyConsume() extends RhoEvent
+
+  final case class DeployTrace(hash: String, events: List[RhoEvent])
+
+  sealed trait Report
+  final case class BlockNotFound(hash: String)                                extends Report
```

Comment:
Should `BlockNotFound` be a `Report`? Perhaps it would make more sense to return `Either[ErrorMessage, Report]` from `trace`?

Reply:
while I understand the motivation I don't agree. There are more possibilities for this return type (introduced in later commits) and these events get parsed to a response. Introducing basically a box without a name will result in the need to remap the box on the presentation layer.
if there were only two options - maybe I could be tempted to solve this with Either but still, this would complicate the presentation layer.
Additionally - it's not really an error. it's a valid state that we can easily handle.